src/Betfair/Adapter/ArrayRpcAdapter.php

```diff
@@ -14,6 +14,21 @@ class ArrayRpcAdapter implements AdapterInterface
     public function adaptResponse($response)
     {
         $data = json_decode($response, true);
-        return $data['result'];
+        
+        // check if $data contains result from Betfair API
+        
+        if(isset($data['result'])){
+            
+        	return $data['result'];
+        	
+        }
+        
+        // check if $data contains error from Betfair API
+        
+        if(isset($data['error'])){
```

Comment:
Hey Daniele,
no worries. I am ok with your corrections.
Also I wanted to ask you a thing: I can't find any BetFair session
management in your classes.... is there a way to save the session token and
reuse it, so to avoid logging in each time the API is called?
I am developing a platform with multiple users using the same API keys, and
Betfair sets a limit to the amount of Login sessions per minute.
Thanks
Giulio

2015-10-26 11:37 GMT+01:00 daniele <notifications@github.com>:

> In src/Betfair/Adapter/ArrayRpcAdapter.php
> <https://github.com/danieledangeli/betfair-php/pull/19#discussion_r42977862>
> :
>
> > @@ -14,6 +14,21 @@ class ArrayRpcAdapter implements AdapterInterface
> >      public function adaptResponse($response)
> >      {
> >          $data = json_decode($response, true);
> > -        return $data['result'];
> > +
> > +        // check if $data contains result from Betfair API
> > +
> > +        if(isset($data['result'])){
> > +
> > +        	return $data['result'];
> > +        	
> > +        }
> > +
> > +        // check if $data contains error from Betfair API
> > +
> > +        if(isset($data['error'])){
>
> As above :D
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/danieledangeli/betfair-php/pull/19/files#r42977862>.
>